File: Data Structures & Algorithms/duplicate-integer/submission-3.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap <Integer, Integer> numCount = new HashMap<>();
        for (int i = 0; i < nums.length; i++){
            if (numCount.containsKey(nums[i])){
                return true;
            }else{
                numCount.put(nums[i], 1 );
            }

        }
        return false;

    }
}
// utilised hashmaps to count numbers - much more efficient compared to brute force method

File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        k = len(nums)
        for i in range(len(nums)):
            if nums[i] == val:
                nums[i] = 101
                k -= 1
        nums.sort()
        print(nums)
        return k

